[Bug]: KafkaConnector with state stopped is not stopped when current connector state is FAILED

### Bug Description

When a KafkaConnector is in FAILED state and the CR is updated to `spec.state: stopped`, Strimzi does not stop the connector.
While most of the time this is not a problem, it can lead to problems where the connector instance itself is in a FAILED state but there are actually still running tasks. These tasks will then not be stopped.

### Steps to reproduce

1. Create a JDBC source connector with a valid configuration which connects to a database
2. Edit the connector and set `connection.password` to a wrong value (Connector instance will now be in FAILED but the old task continues running)
3. Set the state of the connector to stopped
4. Connector will be in FAILED but task will continue to run instead of being stopped

### Expected behavior

I would expect the connector and the tasks to be in `STOPPED` state (or rather the tasks to not exists anymore) after setting the state on the connector to `stopped` even though the connector instance is `FAILED`

### Strimzi version

0.48.0

### Kubernetes version

Kubernetes 1.33

### Installation method

Helm chart

### Infrastructure

Amazon EKS

### Configuration files and logs

```yaml
apiVersion: kafka.strimzi.io/v1beta2
kind: KafkaConnector
metadata:
  creationTimestamp: "2026-03-18T08:45:06Z"
  generation: 15
  labels:
    strimzi.io/cluster: my-cluster
  name: mysql-jdbc
  namespace: mynamespace
  resourceVersion: "104458710"
  uid: cb3187bc-37f0-473e-84d4-80776b2cdea9
spec:
  autoRestart:
    enabled: false
  class: io.confluent.connect.jdbc.JdbcSourceConnector
  config:
    connection.password: wrongPassword # obviously should be mounted from secret, just to show
    connection.url: jdbc:mysql://mysql.mysql:3306/test
    connection.user: root
    key.converter: org.apache.kafka.connect.json.JsonConverter
    mode: timestamp
    table.whitelist: posts
    timestamp.column.name: created_at
    topic.creation.enable: "false"
    topic.prefix: aprefix.
    value.converter: org.apache.kafka.connect.json.JsonConverter
    value.converter.schemas.enable: "false"
  state: stopped
  tasksMax: 1
status:
  conditions:
  - lastTransitionTime: "2026-03-20T10:08:40.734825070Z"
    message: Connector has failed, see connectorStatus for more details.
    reason: Throwable
    status: "True"
    type: NotReady
  connectorStatus:
    connector:
      state: FAILED
      trace: "org.apache.kafka.connect.errors.ConnectException: java.sql.SQLException:
        Access denied for user 'root'@'someip' (using password: YES)\n\tat
        io.confluent.connect.jdbc.util.CachedConnectionProvider.getConnection(CachedConnectionProvider.java:62)\n\tat
        io.confluent.connect.jdbc.JdbcSourceConnector.start(JdbcSourceConnector.java:101)\n\tat
        org.apache.kafka.connect.runtime.WorkerConnector.doStart(WorkerConnector.java:219)\n\tat
        org.apache.kafka.connect.runtime.WorkerConnector.start(WorkerConnector.java:248)\n\tat
        org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:407)\n\tat
        org.apache.kafka.connect.runtime.WorkerConnector.doTransitionTo(WorkerConnector.java:388)\n\tat
        org.apache.kafka.connect.runtime.WorkerConnector.doRun(WorkerConnector.java:168)\n\tat
        org.apache.kafka.connect.runtime.WorkerConnector.run(WorkerConnector.java:128)\n\tat
        org.apache.kafka.connect.runtime.isolation.Plugins.lambda$withClassLoader$1(Plugins.java:254)\n\tat
        java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:539)\n\tat
        java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)\n\tat java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)\n\tat
        java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)\n\tat
        java.base/java.lang.Thread.run(Thread.java:840)\nCaused by: java.sql.SQLException:
        Access denied for user 'root'@'someip' (using password: YES)\n\tat
        com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:121)\n\tat
        com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:114)\n\tat
        com.mysql.cj.jdbc.ConnectionImpl.createNewIO(ConnectionImpl.java:840)\n\tat
        com.mysql.cj.jdbc.ConnectionImpl.<init>(ConnectionImpl.java:416)\n\tat com.mysql.cj.jdbc.ConnectionImpl.getInstance(ConnectionImpl.java:237)\n\tat
        com.mysql.cj.jdbc.NonRegisteringDriver.connect(NonRegisteringDriver.java:180)\n\tat
        java.sql/java.sql.DriverManager.getConnection(DriverManager.java:681)\n\tat
        java.sql/java.sql.DriverManager.getConnection(DriverManager.java:190)\n\tat
        io.confluent.connect.jdbc.dialect.GenericDatabaseDialect.getConnection(GenericDatabaseDialect.java:278)\n\tat
        io.confluent.connect.jdbc.util.CachedConnectionProvider.newConnection(CachedConnectionProvider.java:84)\n\tat
        io.confluent.connect.jdbc.util.CachedConnectionProvider.getConnection(CachedConnectionProvider.java:54)\n\t...
        13 more\n"
      version: 10.9.2
      worker_id: my-cluster-connect-0.my-cluster-connect.mynamespace.svc:8083
    name: mysql-jdbc
    tasks:
    - id: 0
      state: RUNNING
      version: 10.9.2
      worker_id: my-cluster-connect-0.my-cluster-connect.mynamespace.svc:8083
    type: source
  observedGeneration: 15
  tasksMax: 1
  topics:
  - aprefix.posts
  - aprefix.samples
```

### Additional context

_No response_

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,7 @@
 
 ## 1.1.0
 
+* Allow failed `KafkaConnectors` to be stopped and reject pausing of failed connectors since this operation is not supported by Kafka Connect
 * Add support for Apache Kafka 4.3.0 and 4.2.1
 * Remove support for Kafka 4.1.x
 * Reconcile Cruise Control before Entity Operator when both are enabled.

--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
@@ -483,7 +483,7 @@ public abstract class AbstractConnectOperator<C extends KubernetesClient, T exte
             Future<Void> future = Future.succeededFuture();
 
             switch (state) {
-                case "RUNNING" -> {
+                case "RUNNING", "FAILED" -> {
                     if (targetState == ConnectorState.PAUSED) {
                         LOGGER.infoCr(reconciliation, "Pausing connector {}", connectorName);
                         future = Future.fromCompletionStage(apiClient.pause(reconciliation, host, port, connectorName));
@@ -508,15 +508,6 @@ public abstract class AbstractConnectOperator<C extends KubernetesClient, T exte
                     } else if (targetState == ConnectorState.PAUSED) {
                         LOGGER.infoCr(reconciliation, "Pausing connector {}", connectorName);
                         future = Future.fromCompletionStage(apiClient.pause(reconciliation, host, port, connectorName));
-                    }
-                }
-                case "FAILED" -> {
-                    if (targetState == ConnectorState.PAUSED) {
-                        LOGGER.infoCr(reconciliation, "Pausing connector {}", connectorName);
-                        future = Future.fromCompletionStage(apiClient.pause(reconciliation, host, port, connectorName));
-                    } else if (targetState == ConnectorState.STOPPED) {
-                        LOGGER.infoCr(reconciliation, "Stopping connector {}", connectorName);
-                        future = Future.fromCompletionStage(apiClient.stop(reconciliation, host, port, connectorName));
                     }
                 }
                 default -> {

--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
@@ -512,10 +512,7 @@ public abstract class AbstractConnectOperator<C extends KubernetesClient, T exte
                 }
                 case "FAILED" -> {
                     if (targetState == ConnectorState.PAUSED) {
-                        String message = "Connector " + connectorName + " cannot be paused since it is in failed state.";
-                        LOGGER.warnCr(reconciliation, message);
-                        conditions.add(StatusUtils.buildWarningCondition("UpdateConnectorState", message));
-                        return Future.succeededFuture(conditions);
+                        return Future.failedFuture("Connector " + connectorName + " cannot be paused since it is in failed state.");
                     } else if (targetState == ConnectorState.STOPPED) {
                         LOGGER.infoCr(reconciliation, "Stopping connector {}", connectorName);
                         future = Future.fromCompletionStage(apiClient.stop(reconciliation, host, port, connectorName));

--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
@@ -483,7 +483,7 @@ public abstract class AbstractConnectOperator<C extends KubernetesClient, T exte
             Future<Void> future = Future.succeededFuture();
 
             switch (state) {
-                case "RUNNING", "FAILED" -> {
+                case "RUNNING" -> {
                     if (targetState == ConnectorState.PAUSED) {
                         LOGGER.infoCr(reconciliation, "Pausing connector {}", connectorName);
                         future = Future.fromCompletionStage(apiClient.pause(reconciliation, host, port, connectorName));
@@ -508,6 +508,17 @@ public abstract class AbstractConnectOperator<C extends KubernetesClient, T exte
                     } else if (targetState == ConnectorState.PAUSED) {
                         LOGGER.infoCr(reconciliation, "Pausing connector {}", connectorName);
                         future = Future.fromCompletionStage(apiClient.pause(reconciliation, host, port, connectorName));
+                    }
+                }
+                case "FAILED" -> {
+                    if (targetState == ConnectorState.PAUSED) {
+                        String message = "Connector " + connectorName + " cannot be paused since it is in failed state.";
+                        LOGGER.warnCr(reconciliation, message);
+                        conditions.add(StatusUtils.buildWarningCondition("UpdateConnectorState", message));
+                        return Future.succeededFuture(conditions);
+                    } else if (targetState == ConnectorState.STOPPED) {
+                        LOGGER.infoCr(reconciliation, "Stopping connector {}", connectorName);
+                        future = Future.fromCompletionStage(apiClient.stop(reconciliation, host, port, connectorName));
                     }
                 }
                 default -> {

--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
@@ -512,6 +512,7 @@ public abstract class AbstractConnectOperator<C extends KubernetesClient, T exte
                 }
                 case "FAILED" -> {
                     if (targetState == ConnectorState.PAUSED) {
+                        LOGGER.warnCr(reconciliation, "Connector {} cannot be paused since it is in the failed state", connectorName);
                         return Future.failedFuture("Connector " + connectorName + " cannot be paused since it is in failed state.");
                     } else if (targetState == ConnectorState.STOPPED) {
                         LOGGER.infoCr(reconciliation, "Stopping connector {}", connectorName);

--- a/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
+++ b/cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java
@@ -510,6 +510,15 @@ public abstract class AbstractConnectOperator<C extends KubernetesClient, T exte
                         future = Future.fromCompletionStage(apiClient.pause(reconciliation, host, port, connectorName));
                     }
                 }
+                case "FAILED" -> {
+                    if (targetState == ConnectorState.PAUSED) {
+                        LOGGER.infoCr(reconciliation, "Pausing connector {}", connectorName);
+                        future = Future.fromCompletionStage(apiClient.pause(reconciliation, host, port, connectorName));
+                    } else if (targetState == ConnectorState.STOPPED) {
+                        LOGGER.infoCr(reconciliation, "Stopping connector {}", connectorName);
+                        future = Future.fromCompletionStage(apiClient.stop(reconciliation, host, port, connectorName));
+                    }
+                }
                 default -> {
                     // Connectors can also be in the UNASSIGNED or RESTARTING state. We could transition directly
                     // from these states to PAUSED or STOPPED but as these are transient, and typically lead to

--- a/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java
+++ b/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java
@@ -2743,9 +2743,9 @@ public class ConnectorMockTest {
                 eq(connectorName));
     }
 
-    /** Create connect, create connector, simulate FAILED state, pause connector */
+    /** Create connect, create connector in FAILED state, attempt to pause — should log warning and add condition instead of calling pause() */
     @Test
-    public void testConnectorFailedToPaused() {
+    public void testConnectorPauseWhenFailed() {
         String connectName = "cluster";
         String connectorName = "connector";
 
@@ -2782,7 +2782,7 @@ public class ConnectorMockTest {
         waitForConnectorReady(connectorName);
         waitForConnectorState(connectorName, "RUNNING");
 
-        // Simulate connector going into FAILED state by updating the connectors map directly
+        // Simulate connector entering FAILED state
         String host = KafkaConnectResources.qualifiedServiceName(connectName, namespace);
         connectors.put(key(host, connectorName), new ConnectorStatus("FAILED", connectors.get(key(host, connectorName)).config));
 
@@ -2792,12 +2792,9 @@ public class ConnectorMockTest {
                 .endSpec()
                 .build());
 
-        waitForConnectorState(connectorName, "PAUSED");
+        waitForConnectorCondition(connectorName, "Warning", "UpdateConnectorState");
 
-        verify(api, times(1)).pause(any(),
-                eq(KafkaConnectResources.qualifiedServiceName(connectName, namespace)), eq(KafkaConnectCluster.REST_API_PORT),
-                eq(connectorName));
-        verify(api, never()).stop(any(),
+        verify(api, never()).pause(any(),
                 eq(KafkaConnectResources.qualifiedServiceName(connectName, namespace)), eq(KafkaConnectCluster.REST_API_PORT),
                 eq(connectorName));
     }

--- a/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java
+++ b/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java
@@ -2743,7 +2743,7 @@ public class ConnectorMockTest {
                 eq(connectorName));
     }
 
-    /** Create connect, create connector in FAILED state, attempt to pause — should log warning and add condition instead of calling pause() */
+    /** Create connect, create connector in FAILED state, attempt to pause — reconciliation should fail instead of calling pause() */
     @Test
     public void testConnectorPauseWhenFailed() {
         String connectName = "cluster";
@@ -2792,7 +2792,8 @@ public class ConnectorMockTest {
                 .endSpec()
                 .build());
 
-        waitForConnectorCondition(connectorName, "Warning", "UpdateConnectorState");
+        waitForConnectorNotReady(connectorName, "NoStackTraceThrowable",
+                "Connector " + connectorName + " cannot be paused since it is in failed state.");
 
         verify(api, never()).pause(any(),
                 eq(KafkaConnectResources.qualifiedServiceName(connectName, namespace)), eq(KafkaConnectCluster.REST_API_PORT),

--- a/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java
+++ b/cluster-operator/src/test/java/io/strimzi/operator/cluster/operator/assembly/ConnectorMockTest.java
@@ -143,11 +143,11 @@ public class ConnectorMockTest {
         statusNode.put("name", connectorName);
         Map<String, Object> connector = new HashMap<>();
         statusNode.put("connector", connector);
-        connector.put("state", connectorStatus.state.toString());
+        connector.put("state", connectorStatus.state);
         connector.put("worker_id", "somehost0:8083");
         Map<String, Object> task = new HashMap<>();
         task.put("id", 0);
-        task.put("state", connectorStatus.state.toString());
+        task.put("state", connectorStatus.state);
         task.put("worker_id", "somehost2:8083");
         List<Map<String, Object>> tasks = singletonList(task);
         statusNode.put("tasks", tasks);
@@ -289,7 +289,7 @@ public class ConnectorMockTest {
             LOGGER.info("###### create " + host);
             String connectorName = invocation.getArgument(3);
             JsonObject connectorConfig = invocation.getArgument(4);
-            connectors.putIfAbsent(key(host, connectorName), new ConnectorStatus(ConnectorState.RUNNING, connectorConfig));
+            connectors.putIfAbsent(key(host, connectorName), new ConnectorStatus("RUNNING", connectorConfig));
             return CompletableFuture.completedFuture(null);
         });
         when(api.delete(any(), any(), anyInt(), anyString())).thenAnswer(invocation -> {
@@ -318,8 +318,8 @@ public class ConnectorMockTest {
             if (connectorStatus == null) {
                 return CompletableFuture.failedFuture(new ConnectRestException("PUT", "", 404, "Not found", "Connector name " + connectorName));
             }
-            if (!ConnectorState.PAUSED.equals(connectorStatus.state)) {
-                connectors.put(key(host, connectorName), new ConnectorStatus(ConnectorState.PAUSED, connectorStatus.config));
+            if (!"PAUSED".equals(connectorStatus.state)) {
+                connectors.put(key(host, connectorName), new ConnectorStatus("PAUSED", connectorStatus.config));
             }
             return CompletableFuture.completedFuture(null);
         });
@@ -330,8 +330,8 @@ public class ConnectorMockTest {
             if (connectorStatus == null) {
                 return CompletableFuture.failedFuture(new ConnectRestException("PUT", "", 404, "Not found", "Connector name " + connectorName));
             }
-            if (!ConnectorState.STOPPED.equals(connectorStatus.state)) {
-                connectors.put(key(host, connectorName), new ConnectorStatus(ConnectorState.STOPPED, connectorStatus.config));
+            if (!"STOPPED".equals(connectorStatus.state)) {
+                connectors.put(key(host, connectorName), new ConnectorStatus("STOPPED", connectorStatus.config));
             }
             return CompletableFuture.completedFuture(null);
         });
@@ -342,8 +342,8 @@ public class ConnectorMockTest {
             if (connectorStatus == null) {
                 return CompletableFuture.failedFuture(new ConnectRestException("PUT", "", 404, "Not found", "Connector name " + connectorName));
             }
-            if (ConnectorState.STOPPED.equals(connectorStatus.state) || ConnectorState.PAUSED.equals(connectorStatus.state)) {
-                connectors.put(key(host, connectorName), new ConnectorStatus(ConnectorState.RUNNING, connectorStatus.config));
+            if ("STOPPED".equals(connectorStatus.state) || "PAUSED".equals(connectorStatus.state)) {
+                connectors.put(key(host, connectorName), new ConnectorStatus("RUNNING", connectorStatus.config));
             }
             return CompletableFuture.completedFuture(null);
         });
@@ -2684,6 +2684,124 @@ public class ConnectorMockTest {
         assertThat(connectorConfig.getValue().containsKey("connector.plugin.version"), is(false));
     }
 
+    /** Create connect, create connector, simulate FAILED state, stop connector */
+    @Test
+    public void testConnectorFailedToStopped() {
+        String connectName = "cluster";
+        String connectorName = "connector";
+
+        KafkaConnect connect = new KafkaConnectBuilder()
+                .withNewMetadata()
+                    .withNamespace(namespace)
+                    .withName(connectName)
+                    .addToAnnotations(Annotations.STRIMZI_IO_USE_CONNECTOR_RESOURCES, "true")
+                .endMetadata()
+                .withNewSpec()
+                    .withReplicas(1)
+                    .withBootstrapServers("my-kafka:9092")
+                    .withGroupId("my-group")
+                    .withConfigStorageTopic("my-config-topic")
+                    .withOffsetStorageTopic("my-offset-topic")
+                    .withStatusStorageTopic("my-status-topic")
+                .endSpec()
+                .build();
+        Crds.kafkaConnectOperation(client).inNamespace(namespace).resource(connect).create();
+        waitForConnectReady(connectName);
+
+        KafkaConnector connector = new KafkaConnectorBuilder()
+                .withNewMetadata()
+                    .withName(connectorName)
+                    .withNamespace(namespace)
+                    .addToLabels(Labels.STRIMZI_CLUSTER_LABEL, connectName)
+                .endMetadata()
+                .withNewSpec()
+                    .withTasksMax(1)
+                    .withClassName("Dummy")
+                .endSpec()
+                .build();
+        Crds.kafkaConnectorOperation(client).inNamespace(namespace).resource(connector).create();
+        waitForConnectorReady(connectorName);
+        waitForConnectorState(connectorName, "RUNNING");
+
+        // Simulate connector going into FAILED state by updating the connectors map directly
+        String host = KafkaConnectResources.qualifiedServiceName(connectName, namespace);
+        connectors.put(key(host, connectorName), new ConnectorStatus("FAILED", connectors.get(key(host, connectorName)).config));
+
+        Crds.kafkaConnectorOperation(client).inNamespace(namespace).withName(connectorName).edit(spec -> new KafkaConnectorBuilder(spec)
+                .editSpec()
+                .withState(ConnectorState.STOPPED)
+                .endSpec()
+                .build());
+
+        waitForConnectorState(connectorName, "STOPPED");
+
+        verify(api, times(1)).stop(any(),
+                eq(KafkaConnectResources.qualifiedServiceName(connectName, namespace)), eq(KafkaConnectCluster.REST_API_PORT),
+                eq(connectorName));
+        verify(api, never()).pause(any(),
+                eq(KafkaConnectResources.qualifiedServiceName(connectName, namespace)), eq(KafkaConnectCluster.REST_API_PORT),
+                eq(connectorName));
+    }
+
+    /** Create connect, create connector, simulate FAILED state, pause connector */
+    @Test
+    public void testConnectorFailedToPaused() {
+        String connectName = "cluster";
+        String connectorName = "connector";
+
+        KafkaConnect connect = new KafkaConnectBuilder()
+                .withNewMetadata()
+                    .withNamespace(namespace)
+                    .withName(connectName)
+                    .addToAnnotations(Annotations.STRIMZI_IO_USE_CONNECTOR_RESOURCES, "true")
+                .endMetadata()
+                .withNewSpec()
+                    .withReplicas(1)
+                    .withBootstrapServers("my-kafka:9092")
+                    .withGroupId("my-group")
+                    .withConfigStorageTopic("my-config-topic")
+                    .withOffsetStorageTopic("my-offset-topic")
+                    .withStatusStorageTopic("my-status-topic")
+                .endSpec()
+                .build();
+        Crds.kafkaConnectOperation(client).inNamespace(namespace).resource(connect).create();
+        waitForConnectReady(connectName);
+
+        KafkaConnector connector = new KafkaConnectorBuilder()
+                .withNewMetadata()
+                    .withName(connectorName)
+                    .withNamespace(namespace)
+                    .addToLabels(Labels.STRIMZI_CLUSTER_LABEL, connectName)
+                .endMetadata()
+                .withNewSpec()
+                    .withTasksMax(1)
+                    .withClassName("Dummy")
+                .endSpec()
+                .build();
+        Crds.kafkaConnectorOperation(client).inNamespace(namespace).resource(connector).create();
+        waitForConnectorReady(connectorName);
+        waitForConnectorState(connectorName, "RUNNING");
+
+        // Simulate connector going into FAILED state by updating the connectors map directly
+        String host = KafkaConnectResources.qualifiedServiceName(connectName, namespace);
+        connectors.put(key(host, connectorName), new ConnectorStatus("FAILED", connectors.get(key(host, connectorName)).config));
+
+        Crds.kafkaConnectorOperation(client).inNamespace(namespace).withName(connectorName).edit(spec -> new KafkaConnectorBuilder(spec)
+                .editSpec()
+                .withState(ConnectorState.PAUSED)
+                .endSpec()
+                .build());
+
+        waitForConnectorState(connectorName, "PAUSED");
+
+        verify(api, times(1)).pause(any(),
+                eq(KafkaConnectResources.qualifiedServiceName(connectName, namespace)), eq(KafkaConnectCluster.REST_API_PORT),
+                eq(connectorName));
+        verify(api, never()).stop(any(),
+                eq(KafkaConnectResources.qualifiedServiceName(connectName, namespace)), eq(KafkaConnectCluster.REST_API_PORT),
+                eq(connectorName));
+    }
+
     // Utility record used during tests
-    record ConnectorStatus(ConnectorState state, JsonObject config) { }
+    record ConnectorStatus(String state, JsonObject config) { }
 }

